feat: support aggregation over the whole table

jingchen2222 opened the pull request:
* **Please check if the PR fulfills these requirements**

  - The commit message follows our guidelines 
  - Tests for the changes have been added (for bug fixes / features)
  - Docs have been added / updated (for bug fixes / features)


* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature
* **What is the current behavior?** (You can also link to an open issue here)
#219 

* **What is the new behavior (if this is a feature change)?**
Now that we only support aggregation function over a window or group. It makes our users so frustrated since query statements like `SELECT SUM(col1) from t1;` are quite common.
In this issue,  we are going to apply aggregation function over a table to extend the functional ability of OpenMLDB. 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)



* **Other information**:

jingchen2222 committed:
feat: support aggregation over the whole table

jingchen2222 committed:
feat: put restriction on aggregation over table under performance sensitive mode

jingchen2222 committed:
feat: merge main

jingchen2222 committed:
feat: update hybridse java sdk version to 0.2.4-snapshot

aceforeverd commented on hybridse/src/plan/planner.cc:
```diff
@@ -24,7 +24,7 @@
 #include "plan/plan_api.h"
 #include "proto/fe_common.pb.h"
-
+#include "udf/default_udf_library.h"
```
udf required by planner?

jingchen2222 committed:
feat: resolve comment
1. put restriction on table aggregation and row project simultanousily
2. fix unit case
3. support checking isUdaf with given function name

jingchen2222 committed:
feat: update hybridse package

tobegit3hub reviewed (approved):
LGTM
